src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,341 @@
+                addTaskToList(tasks, statusOfSavedTask, newTask);
+                break;
+            case "E":
+                String detailsOfSavedEvent = sectionsOfCurrentLine[3].trim();
+                newTask = new Event(informationOfSavedTask, detailsOfSavedEvent);
+                addTaskToList(tasks, statusOfSavedTask, newTask);
+                break;
+            default:
+                System.out.println("Unknown task type");
+                break;
+            }
+        }
+    }
+
+    private static void addTaskToList(ArrayList<Task> tasks, String statusOfSavedTask, Task newTask) {
+        if (statusOfSavedTask.equals("1")) {
+            newTask.markAsDone();
+        }
+        tasks.add(newTask);
+    }
+
+    private static File checkIfFileExists(String pathOfDataFile) throws IOException {
+        File dataFile = new File(pathOfDataFile);
+        if (!dataFile.exists()) {
+            dataFile.createNewFile();
+        }
+        return dataFile;
+    }
+
+    private static void checkIfDirectoryExists() {
+        File directory = new File("data/");
+        if (!directory.exists()) {
+            directory.mkdir();
+        }
+    }
+
+    private static void saveListToFile(ArrayList<Task> tasks, String pathOfDataFile) throws IOException {
+        writeDataToFile(pathOfDataFile, "Here are the tasks in your list:"
+                                    + System.lineSeparator());
+        for (Task task : tasks) {
+            String description = task.getDescription();
+            String status;
+            if (task.getStatusIcon().equals("\u2713")) {
+                status = "1";
+            } else {
+                status = "0";
+            }
+            String typeOfSaveTask = task.getTypeOfTask();
+            switch (typeOfSaveTask) {
+            case "T":
+                appendDataToFile(pathOfDataFile, typeOfSaveTask + " | "
+                        + status + " | " + description + System.lineSeparator());
+                break;
+            case "D":
+                Deadline deadlineToSave = (Deadline) task;
+                String deadlineDetails = deadlineToSave.getDeadline();
+                appendDataToFile(pathOfDataFile, typeOfSaveTask + " | "
+                        + status + " | " + description + "| " + deadlineDetails + System.lineSeparator());
+                break;
+            case "E":
+                Event eventToSave = (Event) task;
+                String eventDetails = eventToSave.getEventDateTime();
+                appendDataToFile(pathOfDataFile, typeOfSaveTask + " | "
+                        + status + " | " + description + "| " + eventDetails + System.lineSeparator());
+                break;
+            default:
+                System.out.println("Unknown task type");
+                break;
+            }
+        }
+    }
+
+    private static void determineTypeOfTask(ArrayList<Task> tasks, String typeOfTask, String description) {
```

Comment:
Similar for this, I would expect it only to return the type of task, but it actually executes the command.